sources/xfscli/src/bin/xfs_fsck.rs

```diff
@@ -0,0 +1,179 @@
+use std::fs;
+use std::ops::Add;
+use std::os::unix::fs::FileTypeExt;
+use std::path::Path;
+use std::process::exit;
+use std::process::Command;
+use std::str;
+
+use argh::FromArgs;
+
+// This CLI is rewritten in Rust for fsck.xfs shell script
+// https://git.kernel.org/pub/scm/fs/xfs/xfsprogs-dev.git/tree/fsck/xfs_fsck.sh
+// as Shell is not available on a Bottlerocket host.
+
+/// check and repair a XFS file system
+#[derive(FromArgs)]
+struct Args {
+    /// auto repair the file system
+    #[argh(switch, short = 'a')]
+    auto_repair: bool,
+    /// auto check all file systems
+    #[argh(switch, short = 'A')]
+    auto_check: bool,
+    /// force repair filesystem
+    #[argh(switch, short = 'f')]
+    force: bool,
+    /// check the root filesystem as well when -A flag is set
+    #[argh(switch, short = 'p')]
+    p: bool,
+    /// repair filesystem automatically
+    #[argh(switch, short = 'y')]
+    repair: bool,
+
+    /// mount-point | block-device
+    #[argh(positional)]
+    target: Option<String>,
+}
+
+static MOUNT: &str = "/usr/bin/mount";
+static UMOUNT: &str = "/usr/bin/umount";
+static BLKID: &str = "/usr/sbin/blkid";
+static XFS_REPAIR: &str = "/usr/sbin/xfs_repair";
+
+fn main() {
+    let args: Args = argh::from_env();
+    let auto = args.auto_check || args.auto_repair || args.p;
+    let force = args.force;
+    let repair = args.repair;
+
+    if let Some(target) = &args.target {
+        if !Path::new(target).exists() {
+            println!("{} does not exist", target);
+            exit(8);
+        }
+
+        if force {
+            let xfs_repair_result = Command::new(XFS_REPAIR)
+                .args(["-e", target])
+                .output()
+                .expect("Couldn't run xfs_repair for device");
+            println!("Performing XFS repair..");
+            let mut error = xfs_repair_result.status.code().unwrap();
```

Comment:
```suggestion
            let mut repair_exit_code = xfs_repair_result.status.code().unwrap();
```
Is this an error or simply the exit status from running xfs_repair. Error implies something bad happened.